src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      currentState = { ...currentState };
+
+      action.keysToRemove.forEach((key) => {
+        delete currentState[key];
+      });
+    } else if (action.type === 'clear') {
+      currentState = {};
+    }
```

Comment:
Consider using a `switch` statement instead of multiple `if`/`else if` conditions to handle different action types. This aligns with the checklist item that suggests using a `switch` statement for a limited amount of conditions, which makes the code more readable and organized.